zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/migration/MigrationTask.java

```diff
@@ -51,6 +51,14 @@ public interface MigrationTask {
   /** Returns whether the migration needs to run. */
   boolean needsToRun(final MigrationTaskContext context);
 
+  /**
+   * Returns whether the migration is an initialization task, that must be executed even when the
+   * database is empty.
+   */
+  default boolean isInitialization() {
+    return false;
+  }
```

Comment:
💭 To avoid breaking any existing behavior, we could consider defaulting to `true` and explicitly disabling those migration tasks that we've verified are safe to only run when the database is filled. It's just an idea, but I'm not sure whether it's really improving anything. Feel free to do with this what you will.

Reply:
Yes that would be probably the safest path, but then I'm not sure we will actually prioritize this. So if the default will not change, then this PR is effectively introducing no change.
For reference, the need to run migrations only when needed surfaced in this bugfix [PR](https://github.com/camunda/camunda/pull/32388). 
I had a quick look at all the Migrations and none of them seemed "initializations" except for the routing :shrug: